nutanix/resource_nutanix_virtual_machine_clone.go

```diff
@@ -0,0 +1,787 @@
+package nutanix
+
+import (
+	"context"
+	"log"
+	"reflect"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	v3 "github.com/terraform-providers/terraform-provider-nutanix/client/v3"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
+func resourceNutanixVirtualMachineClone() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNutanixVirtualMachineCloneCreate,
+		UpdateContext: resourceNutanixVirtualMachineCloneUpdate,
+		ReadContext:   resourceNutanixVirtualMachineCloneRead,
+		DeleteContext: resourceNutanixVirtualMachineCloneDelete,
+		Schema: map[string]*schema.Schema{
+			"vm_uuid": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
All the common schema should be returned from same function as in future we don't need to make changes at two places.